ScreenBoom/ScreenBoom/View/RightMenuView.swift

```diff
@@ -86,38 +97,91 @@ class RightMenuView: UIView {
                       trailing: trailingAnchor,
                       padding: .init(top: 84, left: 10, bottom: 0, right: 10),
                       size: .init(width: 0, height: 30))
-    playImage.anchor(top: editButton.bottomAnchor,
+    playButton.anchor(top: editButton.bottomAnchor,
                      leading: leadingAnchor,
                      bottom: nil,
                      trailing: trailingAnchor,
                      padding: .init(top: 15, left: 20, bottom: 0, right: 20),
                      size: .init(width: 0, height: 40))
-    pauseImage.anchor(top: playImage.bottomAnchor,
+    pauseButton.anchor(top: playButton.bottomAnchor,
                       leading: leadingAnchor,
                       bottom: nil,
                       trailing: trailingAnchor,
                       padding: .init(top: 10, left: 20, bottom: 0, right: 20),
                       size: .init(width: 0, height: 40))
-    deleteImage.anchor(top: pauseImage.bottomAnchor,
+    deleteButton.anchor(top: pauseButton.bottomAnchor,
                        leading: leadingAnchor,
                        bottom: nil,
                        trailing: trailingAnchor,
                        padding: .init(top: 10, left: 20, bottom: 0, right: 20),
                        size: .init(width: 0, height: 40))
-    shareImage.anchor(top: deleteImage.bottomAnchor,
+    shareButton.anchor(top: deleteButton.bottomAnchor,
                       leading: leadingAnchor,
                       bottom: nil,
                       trailing: trailingAnchor,
                       padding: .init(top: 15, left: 20, bottom: 0, right: 20),
                       size: .init(width: 0, height: 40))
-    eventNameAndCodeButtonLabel.anchor(top: shareImage.bottomAnchor,
+    eventNameAndCodeButtonLabel.anchor(top: shareButton.bottomAnchor,
                                        leading: leadingAnchor,
                                        bottom: bottomAnchor,
                                        trailing: trailingAnchor,
                                        padding: .init(top: 10, left: 20, bottom: 10, right: 20),
                                        size: .init(width: 0, height: 0))
+    
+    addButtonsAction()
+    
+    
   }
   
+  func addButtonsAction() {
+    
+    shareButton.addTarget(self, action: #selector(sideMenuShareButtonPressed(_:)), for: .touchUpInside)
+    
```

Comment:
remove the spaces in between each one. These add target methods could be subbed for a `configureWithAction(action: ()-> Void)` which takes a closure that returns void. so instead of these youd have `shareButton.configureWithAction(){
      self.delegate?.sideMenuShareButtonPressed()
}`

this way you dont have to have all the methods in this class calling the delegate. 

^^ this would go inside the SideMenuButton Subclass I told you about before